dagjose/json_serialization.go

```diff
@@ -318,7 +317,7 @@ func (d *DagJWE) GeneralJSONSerialization() []byte {
 	return result
 }
 
-// Return the flattened json serialization of this JWE
+// FlattenedSerialization return the flattened json serialization of this JWE
 func (d *DagJWE) FlattenedSerialization() ([]byte, error) {
 	jsonRep := d.asJson()
 	jsonRecipient := jsonRep["recipients"].([]map[string]interface{})[0]
```

Comment:
seems like a NPE risk

Reply:
What does NPE mean?

Reply:
Nil pointer exception

Reply:
In particular, I think it may be possible to construct a JWE with no recipients, which would be strange but is valid with respect to the spec so I guess we should accept it. However, it's not possible to produce a flattened serialization for an object with no recipients (what would you put in the `header` and `encrypted_key` attributes). I think what is missing here is an assertion that there is only one recipient, similar to the assertion in `func (d *DagJWS) FlattenedSerialization` that there is only one signature.